test/quantization/pt2e/test_quantize_pt2e_qat.py

```diff
@@ -528,6 +528,58 @@ def forward(self, x):
         example_inputs = (torch.randn(1, 3, 5, 5),)
         self._verify_symmetric_xnnpack_qat_numerics(M(), example_inputs)
 
+    def test_qat_preserve_source_fn_stack(self):
+        """
+        Test whether `source_fn_stack` is preserved after QAT fusion.
+        """
+        class M(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.backbone = TestHelperModules.ConvWithBNRelu(relu=True)
+
+            def forward(self, x):
+                return self.backbone(x)
+
+        # QAT prepare + convert
+        m = M()
+        example_inputs = (torch.randn(1, 3, 5, 5),)
+        quantizer = XNNPACKQuantizer()
+        quantizer.set_global(get_symmetric_quantization_config(is_qat=True))
+        m = capture_pre_autograd_graph(m, example_inputs)
+        m = prepare_qat_pt2e(m, quantizer)
+        m(*example_inputs)
+        m = convert_pt2e(m)
+
+        # Assert these nodes have the right source partitions
+        node_name_to_source_partition = {
+            "_param_constant0": "l__self___backbone_conv",
+            "_param_constant1": "l__self___backbone_conv",
```

Comment:
also node_names are free to change, so we should not rely on it I think

Reply:
sounds good, let me see what I can do